library_generation/cli/entry_point.py

```diff
@@ -0,0 +1,144 @@
+    generate changed libraries based on current generation config with pull
+    request description.
+
+    If baseline generation config is not specified but current generation
+    config is specified, generate all libraries based on current generation
+    config without pull request description.
+
+    If current generation config is not specified but baseline generation
+    config is specified, raise FileNotFoundError because current generation
+    config should be the source of truth of library generation.
+
+    If both baseline generation config and current generation config are not
+    specified, generate all libraries based on the default generation config,
+    which is generation_config.yaml in the current working directory. Raise
+    FileNotFoundError if the default config does not exist.
+
+    The pull request description, if generated, will be available in
+    repository_path/pr_description.txt.
+    """
+    default_generation_config = f"{os.getcwd()}/generation_config.yaml"
+
+    if baseline_generation_config is None and current_generation_config is None:
+        if not os.path.isfile(default_generation_config):
+            raise FileNotFoundError(
+                f"{default_generation_config} does not exist. "
+                "A valid generation config has to be passed in as "
+                "current_generation_config or exist in the current working "
+                "directory."
+            )
+        current_generation_config = default_generation_config
+    elif current_generation_config is None:
+        raise FileNotFoundError(
+            "current_generation_config is not specified when "
+            "baseline_generation_config is specified. "
+            "current_generation_config should be the source of truth of "
+            "library generation."
+        )
+
+    current_generation_config = os.path.abspath(current_generation_config)
+    repository_path = os.path.abspath(repository_path)
+    if not baseline_generation_config:
+        # Execute full generation based on current_generation_config if
+        # baseline_generation_config is not specified.
+        # Do not generate pull request description.
+        generate_from_yaml(
+            config=from_yaml(current_generation_config),
+            repository_path=repository_path,
+        )
+        return
+
+    # Compare two generation configs and only generate changed libraries.
+    # Generate pull request description.
+    baseline_generation_config = os.path.abspath(baseline_generation_config)
+    config_change = compare_config(
+        baseline_config=from_yaml(baseline_generation_config),
+        current_config=from_yaml(current_generation_config),
+    )
+    generate_from_yaml(
+        config=config_change.current_config,
+        repository_path=repository_path,
+        target_library_names=config_change.get_changed_libraries(),
+    )
+    generate_pr_descriptions(
```

Comment:
We probably want to rename `generate_pr_description` to something like `generate_commit_history`, because we just happen to use PR description to capture the commit history, and we could use other ways. It could be a separate PR.